verilog/preprocessor/verilog_preprocess.h

```diff
@@ -87,6 +88,10 @@ struct VerilogPreprocessData {
   // are two separate vectors.
   std::vector<VerilogPreprocessError> errors;
   std::vector<VerilogPreprocessError> warnings;
+
+  // An exit code of 4 bits is used to distinguish between errors met during 
+  // preprocessing.
+  std::bitset<4> exit_code;
```

Comment:
You probably also want an enum that describes the bits:

```c++
enum ResultBitsBits {
     kFileErrorBit = 0;  // file not found or tokenization error
     kMacroUseNotExpandedBit = 1;  // At least one macro could not be expanded
     kIncludeFileNotFoundBit = 2;      // At least one include file could not be found
}
```

I'd probably not even use a `std::bitset`, just a `uint32_t` as this is the code we then exit `main()` with anyway.
Then your enum could be defined as:

```c++
enum ResultBitsBits {
     kFileErrorBit = (1 << 0);  // file not found or tokenization error
     kMacroUseNotExpandedBit = (1 << 1);  // At least one macro could not be expanded
     kIncludeFileNotFoundBit = (1 << 2);      // At least one include file could not be found
}
```
.. then `OR`-ing together the relevant values.
I think the `uint32_t` and `(1 << xxx)` thing would be easier to read to the average reader of the code.

Reply:
I think for me it is easier to read as bitmap and we can use that number right in the return value.

Since it is very common to define these things as bit-patterns directly into `uint32_t` like that is is much quicker to read. Using a `std::bitset<>` would probably slow down readers of the code as they are wondering if there is a deeper meaning to do so.